src/libsync/pushnotifications.h

```diff
@@ -0,0 +1,45 @@
+#ifndef PUSHNOTIFICATIONS_H
+#define PUSHNOTIFICATIONS_H
+
+#include "websocket.h"
+
+namespace OCC {
+
+class Account;
+class AbstractCredentials;
+
+class PushNotifications : public QObject
+{
+    Q_OBJECT
+
+public:
+    // Question: Is it okay to pass a raw account pointer? Because PushNotifications don't really own the account. The PushNotifications
+    // object is owned by the account state. Why not passing AccountState? Maybe would be better but this makes testing more difficult.
```

Comment:
PushNotification being in libsync it can't know about AccountState which kinds of answer that question. :-)